### PR TITLE
Change links from http to https

### PR DESCRIPTION
**Pull request by agiudiceandrea**
In the home page, the following **http** links are changed to **https**:

> Find out more about our tradition of user and contributor meetings!

from [http://blog.qgis.org/category/qgis-hackfest/](http://blog.qgis.org/category/qgis-hackfest/) to [https://blog.qgis.org/category/qgis-hackfest/](https://blog.qgis.org/category/qgis-hackfest/)

> Latest Project Blog Posts (blog.qgis.org)

from [http://blog.qgis.org](http://blog.qgis.org) to [https://blog.qgis.org](https://blog.qgis.org)

> Latest Blog Posts (planet.qgis.org)

from [http://planet.qgis.org](http://planet.qgis.org) to [https://plugins.qgis.org/planet/](https://plugins.qgis.org/planet/)

> Select graphics from The Noun Project collection

from [http://thenounproject.com/](http://thenounproject.com/) to [https://thenounproject.com/](https://thenounproject.com/)

Moreover, the blog posts links in the atom feed of the "Latest Project Blog Posts (blog.qgis.org)" use the **http** protocol. I don't know how to change this incorrect behavior, so I modified the atom function in order to forcibly replace http:// with https:// in the blog posts links.

Side notes:

In the index.html source there is also a link to [**http**://www.osgeo.org](http://www.osgeo.org)
https://github.com/qgis/QGIS-Website/blob/539d2cac1a7831f968a611d3755053e01bc02819/themes/qgis-theme/index.html#L198
but in the on line Website, the link is somehow (how?) already automatically changed to [**https**://www.osgeo.org](https://www.osgeo.org).

The same happens with the link to [**http**://creativecommons.org/licenses/by-sa/3.0/](http://creativecommons.org/licenses/by-sa/3.0/) https://github.com/qgis/QGIS-Website/blob/539d2cac1a7831f968a611d3755053e01bc02819/themes/qgis-theme/page.html#L40
which is already automatically changed to the **https** protocol in the on line Website.

So, I haven't changed them.

### CODE REVIEW

**Comment by DelazJ** [comment]
> but in the on line Website, the link is somehow (how?) already automatically changed to https://www.osgeo.org.

Is it not that OSGeo itself automatically makes the redirection? This is what happens if you type http://osgeo.org (or http://qgis.org); you get the https version.

**Comment by agiudiceandrea** [comment]
> Is it not that OSGeo itself automatically makes the redirection?

I don't think so.

This is the source in GitHub:
https://github.com/qgis/QGIS-Website/blob/36cbbdc60d7a8bf0f76ca6c743689bac7b9671b0/themes/qgis-theme/index.html#L194-L202

and this is what the Web server sends to the browser as html page source:
https://qgis.org/en/site/index.html

```html
<section class="info">
    <div class="container">
        <div class="span6">
            <p>
                <a href="https://www.osgeo.org" class="external">
                    <img width="55%" src="../_static/images/osgeoproject.png"/>
                </a>
            </p>
        </div>
```

As you can see, http://www.osgeo.org is already changed to https://www.osgeo.org directly in the html page source sent by the Web server.

File: themes/qgis-theme/index.html [hunk 1]
```diff
@@ -178,7 +178,7 @@ <h4>{{ _('Upcoming User & Developer Meetings') }}</h4>
             </div>
             <div class="span6">
                 <div class="info-box">
-                    <a href="http://planet.qgis.org" target="_blank">
+                    <a href="https://planet.qgis.org" target="_blank">
```
Comment (agiudiceandrea) [inline themes/qgis-theme/index.html:181]:
```suggestion
                    <a href="https://plugins.qgis.org/planet/" target="_blank">
```

Comment (agiudiceandrea) [inline themes/qgis-theme/index.html]:
```suggestion
                    <a href="https://plugins.qgis.org/planet/" target="_blank">
```

**Comment by DelazJ** [comment]
@agiudiceandrea Do you think we can merge or do you want someone explain you what is happening in the background before we merge? (_I confess I'm also curious_)

**Comment by agiudiceandrea** [comment]
I would like to know how it works, but in the meantime I think this PR could be merged.
It would also be helpful to know who to contact to make sure that the links to the blog posts in the atom feed of the "Latest Project Blog Posts (blog.qgis.org)" use the https protocol "natively", in order to avoid to hack them with https://github.com/qgis/QGIS-Website/pull/802/commits/1f6590d5183bb14b798af8376961973bd8763a1f

I've submitted an issue report https://github.com/qgis/QGIS-Django/issues/85 on QGIS-Django, but it seems it has received no attention till now. Probably it is not the right repository in which file the issue report.

**Comment by rduivenvoorde** [comment]
FYI: we use a 'Feedjack' Django plugin to harvest the feeds, and then make them available as one feed for the website.

Below is current administration, as you see a mix of http and https
I'm not in favour of making all httpS because I just do not know IF all is made httpS.
I'm aware that it is not that difficult nowadays, but still I think there are some old website instances here....:

Screenshot-20200921144546-940x1198

**Comment by mach0** [comment]
I think if we're not going to try it - we will never know.
If some pages fail we can always correct them later ;)

**Comment by agiudiceandrea** [comment]
Hi @mach0 thanks for the interest.
I think the commit 1f6590d5183bb14b798af8376961973bd8763a1f (which forcibly replaces http:// with https:// in the blog posts links) is no longer necessary due to https://github.com/qgis/QGIS-Django/issues/85#issuecomment-771552320 by @timlinux.
So please let me know if you plan to merge this PR: in this case I will revert that commit before the merging if that's okay with you.

**Comment by mach0** [comment]
I thought about checking and merging the pull requests first, and than look into the issues.
Just to have a nice webpage before 3.18 turns into LTR :)

File: themes/qgis-theme/languageswitch.html [hunk 2]
```diff
@@ -269,7 +269,7 @@
                         content = '';
                     }
                     if (addit){
-                        html+='<a class="external" href="'+link+'">';
+                        html+='<a class="external" href="'+link.replace('http://', 'https://')+'">';
```
Comment (agiudiceandrea) [inline themes/qgis-theme/languageswitch.html]:
```suggestion
                        html+='<a class="external" href="'+link+'">';
```

**Pull request merged**